Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/PrimerNolPayProtocol.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  PrimerNolPayProtocol.swift
+//  Debug App Tests
+//
+//  Created by Boris on 4.10.23..
+//  Copyright © 2023 Primer API Ltd. All rights reserved.
+//
+#if canImport(PrimerNolPaySDK)
+import Foundation
+import PrimerNolPaySDK
+
+extension PrimerNolPay: PrimerNolPayProtocol {}
+
+protocol PrimerNolPayProtocol {
+    
+    init(appId: String, isDebug: Bool, isSandbox: Bool, appSecretHandler: @escaping (String, String) async throws -> String)
+    
+    func scanNFCCard(completion: @escaping (Result<String, PrimerNolPayError>) -> Void)
+    
+    func makeLinkingTokenFor(cardNumber: String, completion: @escaping (Result<String, PrimerNolPayError>) -> Void)
```

Comment:
Nit-pick: I would advocate for the style `makeLinkingToken(for cardNumber:...` here. Ill leave it as one comment but the pattern.

Im aware this is a style preference and something we should discuss as a team.

Reply:
renamed all functions.